src/waveform/renderers/allshader/waveformrendermarkrange.h

```diff
@@ -14,17 +13,21 @@ class SkinContext;
 
 namespace rendergraph {
 class GeometryNode;
-}
+} // namespace rendergraph
 
 namespace allshader {
 class WaveformRenderMarkRange;
-}
+} // namespace allshader
 
 class allshader::WaveformRenderMarkRange final : public ::WaveformRendererAbstract,
                                                  public rendergraph::Node {
   public:
     explicit WaveformRenderMarkRange(WaveformWidgetRenderer* waveformWidget);
 
+    void addRange(WaveformMarkRange&& range) {
+        m_markRanges.push_back(std::move(range));
+    }
```

Comment:
avoids the temporary+move, wdyt?
```suggestion
    template<typename Args...>
    void addRange(Args&&... args) {
        m_markRanges.emplace_back(std::forward<Args>(args)...);
    }
```

Reply:
Antoine changed this, I’ll ask him. But why a template?

Reply:
because that is needed for perfect forwarding. Otherwise the compiler can't deduce whether an arg is an lvalue or an rvalue. I recommend you to read Chapter 5 of Scott Meyers Effective Modern C++ which goes into this in detail.

Reply:
I don't have the book at hand.

I don't really like this. I find using a template here for something that isn't really templatized confusing. I prefer slightly less optimised but more clear code 

But wouldn't `m_markRanges.emplace_back(std::move(range));` be enough?

Reply:
that's actually more inefficient unfortunately. I know perfect forwarding is weird, but its a standard pattern required for maximum efficiency.

Reply:
We don't need maximum efficiency here. This is only called once, on construction of the waveform renderers.

Reply:
then keep as is. works for me.